### ISSUE TITLE
No microkernel and bootloader source.

### ISSUE THREAD

**Issue by fxrstor** [FirstBlood12/RedPetyaOpenSSL#1]
Hello,
This repository is the best Petya clone I've seen apart from some others. This has the developer's keys and the encryption is exact replica.
But there is only one exception and that is there is no Petya's microkernel and bootloader source code.
It would be really interesting to see the underlying code of the kernel. Fun fact: there's no kernel source code of Petya on the internet as far I searched.
I request you to add the source code.
Thank you.

**Comment by fxrstor** [comment]
@FirstBlood12

**Comment by FirstBlood12** [comment]
@U65535F Red Petya 16 bit kernel uses Salsa10 or in later versions Salsa20 to encrypt the Master File Table records from 16 bit mode with Salsa20, Salsa20 is implemented in 16-bit mode, it also contains code to calculate number of sectors for the entire MFT table of each NTFS partition to encrypt the MFT sectors from 16-bit mode

**Comment by fxrstor** [comment]
Yes but are you willing to provide the kernel source code? @FirstBlood12

**Comment by FirstBlood12** [comment]
@jsyk57 why do you need the kernel source code? cant you just disassemble it yourself???

**Comment by fxrstor** [comment]
> @jsyk57 why do you need the kernel source code? cant you just disassemble it yourself???

Thats not the objective. I can disassemble petya myself if I want. Will you delete the repo? 
Its for people who dont have time or they dont wanna take the hefty task. That's why I'm asking. @FirstBlood12

**Comment by jsyk57** [comment]
@FirstBlood12

**Comment by fxrstor** [comment]
@FirstBlood12

**Comment by iss4cf0ng** [comment]
Hi @fxrstor ,

I happened to pass by and noticed you were looking for the Petya microkernel/bootloader source code.
If you haven't seen it, you might want to check out this project: [OpenPetya](https://github.com/iss4cf0ng/OpenPetya). This is a re-implementation of Petya.
It might save you some time from doing the hefty disassembly yourself. Hope it helps!

**Comment by fxrstor** [comment]
> Hi @fxrstor ,
> 
> I happened to pass by and noticed you was looking for the Petya microkernel/bootloader source code.
> If you haven't seen it, you might want to check out this project: [OpenPetya](https://github.com/iss4cf0ng/OpenPetya). This is a re-implementation of Petya.
> It might save you some time from doing the hefty disassembly yourself. Hope it helps!

Hey, thanks for commenting. That's awesome.
Great work, I'll check it out.
You earned a star :)
